### DISCUSSION TITLE
How to read or decode M-Bus data

### DISCUSSION THREAD

**Discussion by HolgerAusB** [merbanan/rtl_433#3129] [Q&A]
I just stumbled upon my water-meter and found out, that it has 868MHz radio (Axioma Qalcosonic W1). And I got data with
`-f 868.625M -s 1600k -Y autolevel`
But I think it is encrypted by a key. Because I don't the find the value in JSON, that is shown on the display. The id is the same as on the water-meter, so that is the correct data. I think rtl_433 tries to make the 136-digit string readable and assumes non-encoded data.
- So how can I decode the data, after getting a key from my water provider? Preferably without installing an additional git repo.
- what are the key names stand for?
- I would like to read the total consumption in m<sup>3</sup>, the leak detection, burst detection and maybe the battery status.

Here is an example. I got two different datasets, which are alternating. I was masking/changing some data for security/privacy reasons:

```json
{
    "time": "1735570773",
    "protocol": 104,
    "model": "Wireless-MBus",
    "mode": "T",
    "M": "AXI",
    "id": 1234567,
    "version": 16,
    "type": 7,
    "type_string": "Water",
    "C": 68,
    "data_length": 68,
    "data": "434...136_digits_total...1ae",
    "mic": "CRC",
    "CI": 123,
    "AC": 0,
    "ST": 0,
    "CW": 0,
    "inst_volume_flow_min_1": "2345.012 m3/min",
    "max_energy_wh_30": "0.000 Wh",
    "min_opertime_m_0": "234567.000 min",
    "max_energy_j_27": "43210.100 kJ",
    "min_energy_wh_0": "0.000 MWh",
    "max_power_jh_0": "0.000 MJ/h",
    "err_power_jh_13": "0.000 MJ/h",
    "mod": "FSK",
    "freq1": 868.88749,
    "freq2": 868.98842,
    "rssi": -11.4623,
    "snr": 20.26793,
    "noise": -31.7303
}

{
    "time": "1735570790",
    "protocol": 104,
    "model": "Wireless-MBus",
    "mode": "T",
    "M": "AXI",
    "id": 1234567,
    "version": 16,
    "type": 7,
    "type_string": "Water",
    "C": 68,
    "data_length": 68,
    "data": "434...136_digits_total...1f3",
    "mic": "CRC",
    "CI": 123,
    "AC": 0,
    "ST": 0,
    "CW": 0,
    "inst_volume_flow_min_1": "2345.012 m3/min",
    "max_energy_wh_30": "0.000 Wh",
    "min_opertime_m_0": "234567.000 min",
    "max_hca_0": "6543.000 ",
    "inst_power_w_0": "0.000 W",
    "mod": "FSK",
    "freq1": 868.88819,
    "freq2": 868.99123,
    "rssi": -11.3307,
    "snr": 17.1962,
    "noise": -28.5269
}
```
Hint for others. Don't try this on weekends or at evening/night. The battery lasts 16 years, so it only sends data during daytime Mondays to Fridays. But that depends on the configuration of the water provider

**Comment by zuckschwerdt** [comment]
Also try the branch from PR #2997 -- it needs to be finalized but there are some changes and fixes to meter readings.

**Comment by ghost** [comment]
good luck, getting the key! lol
And even with a working key- rtl_433 cant decode it yet and gives "encrypted Data" Error Message.
more Work for Christian..
But a lot of M-Bus devices are Not fully encrypted.
Have fun.

I dont think your device is encrypted.
CW is 0, !
Copy the data part of the JSON Output and drop it in wmbusmeters Code Input field.
Christian hast updated a lot of code, so try the branch from PR #2997. Most parts work. Smoke detection is broken, but who needs this..
Beside this, OMS Mode "T" is very old and should Not be used since 2017. Mode C and S is up to date.
Boing

**Comment by merbanan** [comment]
There is another project that handles lots of more wmbus devices including decryption support. Look into https://github.com/wmbusmeters/wmbusmeters.

**Reply by ghost** [reply]
thx Benjamin, ofc i know this project and send a lot of W-M Bus messages to verify.
have a good new year
Boing
